ExtraDry/ExtraDry.Core/Slug.cs

```diff
@@ -202,6 +213,9 @@ private static string ToSlugInternal(string name, bool lowercase = true, string
             }
         }
         name = sb.ToString();
+        if(name.All(c => c == '-')) {  // If the name is composed of only invalid characters, generate a random slug replacement.
```

Comment:
I don't think this belongs here, this should work with the ToUnique*** methods which can check for existing values.  While the chance of collision is quite small, it is not zero.

Reply:
I've moved this to the ToUnique functions. Have a look, there is a slight change in behaviour. If you're asking for a unique slug, having an empty string returned is no longer an option. An empty slug would be invalid and this felt like the right time to solve that.